components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Discovery/src/Services/DiscoveryServices.cs

```diff
@@ -512,14 +518,20 @@ public class DiscoveryServices : IDiscoveryServices, IScannerServices, IDisposab
         /// <returns></returns>
         private async Task AddLoopbackAddressesAsync(List<IPAddress> addresses) {
             // Check local host
+            string hostName = Environment.GetEnvironmentVariable(IoTEdgeVariables.IOTEDGE_GATEWAYHOSTNAME);
             try {
                 if (Host.IsContainer) {
                     // Resolve docker host since we are running in a container
-                    var entry = await Dns.GetHostEntryAsync(kDockerHostName);
+                    if (string.IsNullOrEmpty(hostName)) {
+                        _logger.Information("Gateway Host Name not set");
+                        return;
+                    }
+                    _logger.Debug("Resolve IP for Gateway Host Name: {address}", hostName);
+                    var entry = await Dns.GetHostEntryAsync(hostName);
                     foreach (var address in entry.AddressList
                                 .Where(a => a.AddressFamily == AddressFamily.InterNetwork)
                                 .Where(a => !addresses.Any(b => a.Equals(b)))) {
-                        _logger.Information("Including host address {address}", address);
+                        _logger.Information("Include gateway host address {address}", address);
```

Comment:
I'm wondering if "Including" would be a better word. With "Include" I'm not sure what is happening.

Reply:
thanks! done